### ISSUE TITLE
Simplify controller architecture, merge "operator" and "controller"

### ISSUE THREAD

**Issue by ericavonb** [openshift/machine-config-operator#878]
The operator controller for deploying the core controller is an extraneous layer. The "controller" _is_ the operator, it handles the operational logic specific to this operator (the machine configuration). Other operators exist to handle deploying the operators themselves (CVO in this case,.

**Comment by cgwalters** [comment]
I agree with this - although with this type of thing I have a temptation to ping Abhinav to understand more of his reasoning, but OTOH he's pretty busy with things like reviewing a gigantic PR to add baremetal to the installer, etc.

Probably makes the most sense to approach this in incremental phases; the fact that we have just one image now is a big one!  Next would be to run the operator and controller in the same pod probably.

After that we can start to take advantage of that and drop e.g. the `controllerconfig` CRD which I think is only used as a way to sync from the operator :arrow_right: controller...we could just use a yaml file in the pod shared memory or whatever.

And then perhaps after that we could link the controller's code into the operator process.

**Comment by abhinavdahiya** [comment]
I disagree with operator is the controller.

The controller + server + daemon is the ecosystem that realizes the machineconfiguration API, provides a way to manage configuration of machines in the cluster.

The operator is the one that knows information about how to configure them and update them through versions.

For example, 
configpool is transparent to controller, it updates the nodes with new configuration with max unavail set. The operator knows a configpool is for control plane so it should be set to something safe.

Or that when upgrading it's required that control plane upgrades while the compute ones can trail..

I think the management and control-loop layer separation is useful.
